aips/aip-resource-access.md

```diff
@@ -0,0 +1,196 @@
+---
+aip: 
+title: Resource Access Control
+author: wg@aptoslabs.com, brian.murphy@aptoslabs.com, runtian@aptoslabs.com
+discussions-to (*optional): TODO
+Status: Draft 
+last-call-end-date (*optional): <mm/dd/yyyy TBD>
+type: Standard
+created: <10/11/2023>
+---
+
+# AIP-X - Resource Access Control
+  
+## Summary
+
+In Move, functions can access arbitrary resources as long as they have access to public APIs allowing to manipulate the resource. There are multiple problems with this, as discussed in the [Motivation](#Motivation) section. This AIP proposes an extension to the Move language which allows for fine-grained access control of resources. This is achieved by generalizing the familiar `acquires T` declaration in Move in a downwards compatible way. The evaluation of the resulting access control discipline is primarily _dynamic_, but intended to become static in the future. The dynamic-first approach is natural because the redundancy principle of "paranoid" VM mode makes a dynamic check necessary anyway.
```

Comment:
Also, is "paranoid" VM mode documented somewhere?  If not, it could perhaps use more explanation here.

Reply:
> "downwards"? Perhaps you mean "backwards".

They are synonyms. You can Google it.

Reply:
> Also, is "paranoid" VM mode documented somewhere? If not, it could perhaps use more explanation here.

We have blogs about paranoid mode ("Securing Move"), so yes I think this is sufficiently well known.

Reply:
I think it comes from my 10 years at Microsoft, they often use different notions than the mainstream. BTW I can happily change this if it is better English (which I'm certainly no expert in), I was just defending that the notion in fact exists as a synonym.